http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -404,6 +445,16 @@ object LeonardoJsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
     }
   }
 
+  implicit object JupyterDockerImageJsonFormat extends JsonFormat[JupyterDockerImage] {
+    def read(json: JsValue): JupyterDockerImage = json match {
+      case JsString(imageUrl) =>
+        JupyterDockerImage.stringToJupyterDockerImage(imageUrl)
+          .getOrElse(throw DeserializationException(s"Invalid docker registry. Only ${ContainerRegistry.allRegistries.mkString(", ")} are supported"))
```

Comment:
Does `DeserializationException` here result in a 400 Bad Request status code?

Reply:
I need to test this out. But I'll be surprised if that's not the case :)

Reply:
```curl -v -X PUT --header 'Content-Type: application/json' --header "Authorization: Bearer `gcloud auth print-access-token`" -d '{"jupyterDockerImage": "us.d/broad-dsp-gcr-public/terra-jupyter-hail:0.0.1"}' http://127.0.0.1:8080/qi
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> PUT /qi HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/7.54.0
> Accept: */*
> Content-Type: application/json
> Authorization: Bearer ya29.ImGIB7VHi3-APJWcDb6wV-jZSL7aKhnjXkTla2JdjFCcfxVZdus-ubYQfNo3JivvpE_wFMUBLpD44dAWkDYpCDK4JazhwD_za7I2rpZVQeBoPvutnVSboae8Cb_a-U8oMTVB
> Content-Length: 76
>
* upload completely sent off: 76 out of 76 bytes
< HTTP/1.1 400 Bad Request
< Server: akka-http/10.1.8
< Date: Thu, 19 Sep 2019 02:28:52 GMT
< Content-Type: text/plain; charset=UTF-8
< Content-Length: 94
<
The request content was malformed:
* Connection #0 to host 127.0.0.1 left intact
Invalid docker registry. Only gcr, docker hub are supported%```